nav2_costmap_2d/test/unit/declare_parameter_test.cpp

```diff
@@ -68,7 +68,7 @@ TEST(DeclareParameter, useInvalidParameter)
   try {
     layer.declareParameter("test2", rclcpp::PARAMETER_STRING);
     FAIL() << "Incorrectly handling test_layer.test2 parameter which was not set";
-  } catch (rclcpp::exceptions::NoParameterOverrideProvided & ex) {
+  } catch (rclcpp::exceptions::ParameterUninitializedException & ex) {
```

Comment:
`declare_parameter` cannot throw `ParameterUninitializedException`, I'm not sure if `declareParameter` is trying using some "declare or get" logic, or sth like that.

Reply:
it seems to be doing that https://github.com/ros-planning/navigation2/blob/30b405c58e6d53ba8c96381416bc4679d35a1483/nav2_costmap_2d/src/layer.cpp#L88

the change looks okay to me then

Reply:
Thanks, nice catch. I'll fix this.

Reply:
I've moved the exception handling to the place where `get_parameter` is called (8c44967). Though, it's not clear to me how this test was passing originally since it looks like the exception was not being propagated though `Layer::declareParameter`. I'll do some more testing locally.

Reply:
Nevermind, I was confused. I think this fixes the test (8a33fc2)